build/lib/textextraction/extractors.py

```diff
@@ -0,0 +1,166 @@
+import logging
+import os
+import re
+import subprocess
+
+"""
+The functions below are minimal Python wrappers around Ghostscript, Tika, and
+Tesseract. They are intended to simplify converting pdf files into usable text.
+"""
+
+
+class TextExtraction:
+    """ The TextExtraction class contains functions for extracting and saving
+    metadata and text from all files compatible with Apache Tika"""
+
+    def __init__(self, doc_path, tika_port=9998):
+
+        self.doc_path = doc_path
+        self.root, self.extension = os.path.splitext(doc_path)
+        self.tika_port = tika_port
+        self.text_arg_str = 'curl -T {0} http://localhost:{1}/tika' + \
```

Comment:
Should specify the URL for the tika servers somewhere else? As a class variable perhaps? It seems weird to have that hard-coded here.